REFERENCE.md

```diff
@@ -425,6 +426,35 @@ def assignees_modal
 end
 ```
 
+### Table of Contents
+
+Based on the provided [scrollspy][m-scrollspy] and [pushpin][m-pushpin] methods
+the MTL table of content behaviour automatically creates a dynamic ToC, based on
+the current page content.
+
+A ToC entry is indicated by adding `data-mtl-section="Title"` and then is
+automatically added in the order of appearance on the site to the
+`ul[data-mtl-toc]` element. Any content within `data-mtl-toc` is completely
+replaced.
+
+```html
+<div class="row" data-mtl-section="Introduction">
+  <!-- content -->
+</div>
+<div class="row" data-mtl-section="Structure">
+  <!-- content -->
+</div>
+<div class="row" data-mtl-section="Other">
+  <!-- content -->
+</div>
+
+<!-- ToC element: an empty ul with class="table-of-contents" -->
+<ul class="table-of-contents" data-mtl-toc="pin"></ul>
```

Comment:
Was thinking about this too, but wanted to separate styling from behaviour - so if you have `class="my-cool-toc"` as a class it would still work and not do anything one doesn't want or expect, but I'm open...